tests/test_meltingpot.py

```diff
@@ -0,0 +1,57 @@
+"""Tests the functionality of the MeltingPotCompatibility wrapper on meltingpot substrates."""
+
+import numpy as np
+import pytest
+
+from shimmy.meltingpot_compatibility import MeltingPotCompatibilityV0
+from meltingpot.python.configs.substrates import SUBSTRATES
+from pettingzoo.test import parallel_api_test
+
+
+@pytest.mark.parametrize("substrate_name", SUBSTRATES)
+def test_passing_substrates(substrate_name):
+    """Tests the conversion of all openspiel envs."""
+    for _ in range(5):
+        env = MeltingPotCompatibilityV0(substrate_name=substrate_name, render_mode=None)
+
+        # api test the env
+        parallel_api_test(env)
+
+        env.reset()
+        while env.agents:
+            actions = {agent: env.action_space(agent).sample() for agent in env.agents}
+            observations, rewards, terminations, truncations, infos = env.step(actions)
+
+
+@pytest.mark.parametrize("substrate_name", SUBSTRATES)
+def test_seeding(substrate_name):
+    """Tests the seeding of the openspiel conversion wrapper."""
+    # load and convert the envs
+    env1 = MeltingPotCompatibilityV0(substrate_name=substrate_name, render_mode=None)
+    env2 = MeltingPotCompatibilityV0(substrate_name=substrate_name, render_mode=None)
+
+    env1.reset(seed=42)
+    env2.reset(seed=42)
+
+    a_space1 = env1.action_space(env1.agents[0])
+    a_space1.seed(42)
+    a_space2 = env2.action_space(env1.agents[0])
+    a_space2.seed(42)
+
+    while env1.agents:
+        actions1 = {agent: env1.action_space(agent).sample() for agent in env1.agents}
```

Comment:
Are all of the spaces shared otherwise this isn't deterministic.
@jjshoots Should each agent's space be a unique object?

Reply:
The way that PettingZoo is built, yes, each agent's space is a unique object. However, if we want to use a shared object for all of them, the way to do it is by making the function return the same action space regardless of the argument passed in. I believe this is already done [here](https://github.com/Farama-Foundation/Shimmy/pull/39/files/babbb09e7f93eb0a0005e70d322ffba9498c7829#diff-82ddd83028a0705b182c4f23382fb4b809c549d2ec6f694b21875b5de374828bR107).

Reply:
Not sure what the best way to do this is, I can change it to use separate action spaces for each agent if that’s better. The previous melting pot code did the same thing as I’m doing now but I can see a rationale for making it separate per agent potentially. Would it be non deterministic because of how the functools caching works? I feel like since that’s the way I’ve seen it made consistent for over envs it probably makes sense to keep the action spaces the same. Just makes things simpler.

Reply:
Honestly if meltingpot has the same action spaces for each agent, this method is probably fine. I think an even faster way would be to do it [how KAZ does it](https://github.com/Farama-Foundation/PettingZoo/blob/master/pettingzoo/butterfly/knights_archers_zombies/knights_archers_zombies.py#L380). We eliminate the need for functools then. If all agents indeed share the same action space, we don't even need the dictionary, just return the same action space for every agent.

Reply:
Defining them all in the constructor ahead of time sounds like a good idea to me although @pseudo-rnd-thoughts and you should make the decision I think. Is it faster to use dict(zip()) than using list comprehension in a dict? Style wise I feel like it's a little complex and not immediately obvious how it works with the unused variable _ and enumerate normally returning two values, etc. 
[KAZ:](https://github.com/Farama-Foundation/PettingZoo/blob/master/pettingzoo/butterfly/knights_archers_zombies/knights_archers_zombies.py#L380)
```
self.action_spaces = dict(
            zip(self.agents, [Discrete(6) for _ in enumerate(self.agents)])
        )
```
Dict comprehension:
```
self.action_spaces = {agent: Discrete(6) for agent in self.agents}
```

Reply:
Given that this is in the init, I doubt it will affect things too much as this is usually only called once per training session. That said, I agree that the second choice is much cleaner and we should probably do that.

Reply:
To me, I thought the ability for each individual agent to have their own unique action space allowing deterministic sampling for each agent would be a helpful property. That was the motivation for me with the proposal. 
@jjshoots Knows PZ better than I so I would follow his suggestion